### ISSUE TITLE
[Suggestion] Buying without registration

### ISSUE THREAD

**Issue by Zinnorov** [Azuriom/Plugin-Shop#155]
It will be nice to have the option of buying stuff without the registration. Putting a nickname from the server in the input field is a lot easier than wasting time with registration/logging in.

**Comment by HC-Emro** [comment]
J'aimerais bien aussi, merci.

**Comment by TabZz147** [comment]
how i can activate "buying without registration" ?

**Comment by Futuraura** [comment]
It hasn't been added yet, but I +1 to this, I think it will really boost sales.

**Comment by MrMicky-FR** [comment]
Please use a 👍 reaction instead of a comment when voting for suggestions. Also by using Microsoft or Steam login, or automatic registration, players won’t have to even register so friction is minimal

**Comment by Futuraura** [comment]
> Please use a 👍 reaction instead of a comment when voting for suggestions. Also by using Microsoft or Steam login, or automatic registration, players won’t have to even register so friction is minimal

Not all servers have licensed players only (online-mode=false), so it does make it tedious to explain to the whole player base that they already have an account or they have to register to donate to you :/
Also, it's bad because it gives them time to think about the purchase and also it kind-of acts a barrier to stop them from buying something.

**Comment by MrMicky-FR** [comment]
> > Please use a 👍 reaction instead of a comment when voting for suggestions. Also by using Microsoft or Steam login, or automatic registration, players won’t have to even register so friction is minimal
> 
> Not all servers have licensed players only (online-mode=false), so it does make it tedious to explain to the whole player base that they already have an account or they have to register to donate to you :/ Also, it's bad because it gives them time to think about the purchase and also it kind-of acts a barrier to stop them from buying something.

When using offline-mode, you must have an auth pkugin on your server. In this case you can synchronize it with Azuriom, so the accounts are automatically created on the website when the user registers in-game. When using AzLink or LimboAuth, this can be enabled in the AzLink `config.yml`

**Comment by Zinnorov** [comment]
Yeah, but logging into the website to buy something is a completely different story. People are lazy and don’t realize they can log in with their AuthMe info. Of course, you could mention it at spawn or elsewhere, but most people will just ignore it—it’s too time-consuming. That’s not eBay or Amazon.

**Comment by SwloBr** [comment]
I'd like a similar function, because it would be incredible if you went to the checkout and when it came time to pay, you had to create an account to continue, due to the fact that creating an account beforehand obligatorily ends up making the user not want to make the purchase, so if you don't think it's decent not to let people buy without an account, at least put in this function, because people end up abandoning the idea of buying because it's necessary to log in, so doing this, forcing them to create an account at checkout, retains customers much more

**Comment by MrMicky-FR** [comment]
Summarizing again due to the new messages:
* For Steam and Microsoft login: the **registration is automatic**, and the only required step is a login
* For Minecraft in offline mode, several different auth plugins are supported in AzLink to enable auto-registration. When enabled, a new user on the server is **automatically registered on the website**, with the **same credentials**

Also the issue is open so still planned, but the implementation is not easy to do, **no need to bump and please use a 👍 reaction instead**. The plugin is open-source, so a clean pull request is welcome 🥇

**Comment by Futuraura** [comment]
Just wanted to bring some fresh air into this — while yes, Microsoft/Steam/auto-reg options exist, they still assume a level of tech understanding from the user that simply isn't there for many younger or casual players. A huge portion of Minecraft players are kids or teens who don't fully get account linkage, and if you add any extra steps between wanting to buy and the actual checkout, you risk killing conversion.

This isn't about "how easy it technically is" — it's about how it feels to the average user. And to many, the second they see “register” or “log in,” that's an immediate exit. No other monetization platform forces you into an account before even seeing the checkout page or the cart. Even G2A and humble bundle let you buy before asking for accounts.

A simple accountless checkout, where a player just drops in their nickname & email then pays, could boost sales massively, especially for small or casual servers who don't need the website for anything else, but the ranks and maybe skins.

Even if implementation is complex, maybe this could be a toggle-able option with a warning like “not recommended for certain setups” or “only works with X delivery method.” That way people can choose based on their needs.

This change is about removing friction, not changing how Azuriom works for everyone — just giving us flexibility.

**Comment by MrMicky-FR** [comment]
Hey! Exciting updates 🎉 

Guest purchases are now live on the [`dev/guest-purchases`](https://github.com/Azuriom/Plugin-Shop/tree/dev/guest-purchases) branch and will be rolling out this week or next (just waiting till a few themes are adapted). Steam isn't supported since usernames aren't unique and can contain problematic characters.

This **first step** requires users to be registered before purchasing. Would this work for your use case? For Minecraft servers with auto-registration through auth plugins and AzLink, this system is **completely user-friendly**: customers just enter their username, no technical knowledge required.

Regarding your points:

> This isn't about "how easy it technically is" 
> This change is about removing friction, not changing how Azuriom works for everyone — just giving us flexibility.

> No other monetization platform forces you into an account before even seeing the checkout page or the cart.

You're right about reducing friction! Technical complexity does matter though, with 100+ Azuriom extensions, important changes need careful implementation.

Let's compare what can be compared. Most monetization platforms are solely monetization platforms, not complete integrated website systems with dozens of extensions where registration is required. Also, most are SaaS with no self-hosting or advanced code customization through plugins where you can literally do anything (and 0% fees as a bonus). Both have pros and cons.

As a reminder, Azuriom and the shop plugin are open-source, so pull requests are welcome!

What do you think about this approach? Any specific scenarios where we'd need to iterate further?

Thanks!

**Comment by Futuraura** [comment]
> For Minecraft servers with auto-registration through auth plugins and AzLink, this system is completely user-friendly: customers just enter their username, no technical knowledge required.

This _is_ the perfect way to make loginless purchases a thing. I can't wait to see this feature go live on the main branch.
Thank you, MrMicky!
